k4EDM4hep2LcioConv/src/k4EDM4hep2LcioConv.cpp

```diff
@@ -53,9 +53,6 @@ std::optional<int32_t> attachParticleIDMetaData(IMPL::LCEventImpl* lcEvent, cons
   if (!pidMetaInfo.has_value()) {
     return std::nullopt;
   }
-  if (pidMetaInfo.has_value() && !recoName.has_value()) {
-    return pidMetaInfo->algoType();
-  }
```

Comment:
why is this removed in this PR? It seems to be unrelated to the rest.

Reply:
Ah now I remember. I noticed this is impossible to happen because if `!recoName.has_value()` then the function returns earlier. I can take it out of this PR

Reply:
wondering what my original intention was with this. It almost looks like it might need to appear as a first check before the early return with `!recoName.has_value()`. Maybe let's take it out of this one and follow up in a separate one.